IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -390,5 +453,47 @@ + (NSString *)getPrimaryEccTenantWithSharedAccessGroup:(NSString *)sharedAccessG
     return res;
 }
 
-
++ (MSIDWPJMetadata *)readWPJMetadataWithSharedAccessGroup:(NSString *)sharedAccessGroup
+                                      tenantIdentifier:(NSString *)tenantIdentifier
+                                            domainName:(NSString *)domainName
+                                               context:(id<MSIDRequestContext>)context
+                                                 error:(NSError **)error
+{
+    NSMutableDictionary *query = [NSMutableDictionary new];
+    query[(id)kSecClass] = (id)kSecClassGenericPassword;
+    query[(id)kSecAttrService] = tenantIdentifier;
+    query[(id)kSecAttrAccount] = domainName;
+    query[(id)kSecAttrAccessGroup] = sharedAccessGroup;
+    query[(id)kSecReturnAttributes] = (id)kCFBooleanTrue;
+    query[(id)kSecReturnData] = (id)kCFBooleanTrue;
+    
+    CFDictionaryRef attributeDictCF = NULL;
+    OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query,(CFTypeRef *)&attributeDictCF);
+    
+    NSDictionary *attributeDictionary = CFBridgingRelease(attributeDictCF);
+    NSData *metadataBlob = [attributeDictionary objectForKey:(__bridge id)kSecValueData];
+    
+    NSError *subError = nil;
+    NSDictionary *decodedDataDict = [NSJSONSerialization JSONObjectWithData:metadataBlob
```

Comment:
should we use deserialize api from MSIDJsonSerializer ?

Reply:
Good point, addressed.